riscv-executor/src/lib.rs

```diff
@@ -280,6 +319,9 @@ mod builder {
         /// Current memory.
         mem: HashMap<u32, u32>,
 
+        /// Separate register memory.
+        reg_mem: HashMap<u32, Elem<F>>,
```

Comment:
Any reason not to use a fixed-size array? I think this this would also simplify some code (you can always assume the key is set, values are zero-initialized, ...).

Reply:
It's a bit stricter but sure, will change.

Reply:
actually a fixed-sized array is tricky, because we don't quite know how many registers we're using. There's the 32 RISCV ones, but then the random `tmp*` we use sometimes. I'd prefer to do that in a follow up PR.